gslib/tests/test_Doption.py

```diff
@@ -35,6 +36,44 @@
 class TestDOption(testcase.GsUtilIntegrationTestCase):
   """Integration tests for gsutil -D option."""
 
+  def assert_header_in_output(self, name, value, output):
+    """Asserts that httplib2's debug logger printed out a specified header.
+
+    This method is fairly primitive and uses assertIn statements, and thus is
+    case-sensitive. Values should be normalized (e.g. to lowercase) if
+    capitalization of the expected characters may vary.
+
+    Args:
+      name: (str) The header name, e.g. "Content-Length".
+      value: (Union[str, None]) The header value, e.g. "4096". If no value is
+          expected for the header or the value is unknown, this argument should
+          be `None`.
+      output: (str) The string in which to search for the specified header.
+    """
+    expected = 'header: %s:' % name
+    if value:
+      # Only append a space and then the header value if a value was expected.
+      expected += ' %s' % value
+    if expected in output:
+      return
+
+    # Try the other format - when sending requests via the XML API, headers are
+    # printed in a list of 2-tuples (by Boto), so we test for that output style
+    # as well.  The above style is generally preferred, but Python's http client
+    # doesn't print all values in scenarios where a header is sent multiple
+    # times with different values, e.g. in this case:
+    #   x-goog-hash: md5=blah2
+    #   x-goog-hash: crc32c=blah1
+    # the debug logger would just print the last one to occur (the crc32c hash).
+    alt_expected = "('%s'" % name
```

Comment:
Nit: Instead of having an assignment with an if in the middle, could we break the if outside the assignment and do the string formats separately? A little hard to read currently imo, but if this is the most succinct way I don't mind it